internal/provider/resource_managed_identity_access_rules.go

```diff
@@ -205,36 +200,34 @@ func (t *managedIdentityAccessRuleResource) Read(ctx context.Context,
 		return
 	}
 
-	// Copy the from-Tharsis run stage to the state.
-	state.RunStage = types.StringValue(string(found.RunStage))
+	if found != nil {
+		// Copy the from-Tharsis run stage to the state, but not if it no longer exists.
```

Comment:
It looks like this resource is handling the not found condition differently than the other resources. There should be a nil check to determine if the resource wasn't found instead checking the error string returned from the SDK.

Reply:
This module has a nil check on line 203.  I added it after hitting a panic from one of the other resources.  It might be redundant, or it might not.  Can you please clarify so I can determine in which direction to make the fix?